provider/pkg/provider/helm_release.go

```diff
@@ -250,11 +249,37 @@ func (r *helmReleaseProvider) getActionConfig(namespace string) (*action.Configu
 
 func decodeRelease(pm resource.PropertyMap) (*Release, error) {
 	var release Release
+	values := map[string]interface{}{}
 	stripped := pm.MapRepl(nil, mapReplStripSecrets)
 	logger.V(9).Infof("Decoding release: %#v", stripped)
+
+	if pm.HasValue("valueYamlFiles") {
+		v := stripped["valueYamlFiles"]
+		switch reflect.TypeOf(v).Kind() {
+		case reflect.Slice, reflect.Array:
+			s := reflect.ValueOf(v)
+			for i := 0; i < s.Len(); i++ {
+				val := s.Index(i).Interface()
+				switch t := val.(type) {
+				case *resource.Asset:
+					b, err := t.Bytes()
+					if err != nil {
+						return nil, err
+					}
+					if err = yaml.Unmarshal(b, &values); err != nil {
+						return nil, err
+					}
+				default:
+					return nil, fmt.Errorf("unsupported type for 'valueYamlFiles' arg: %T", v)
+				}
+			}
+		}
+	}
+
 	if err := mapstructure.Decode(stripped, &release); err != nil {
 		return nil, fmt.Errorf("decoding failure: %w", err)
 	}
+	release.Values = mergeMaps(release.Values, values)
```

Comment:
The precedence - if it's not clear - will prioritize the values in the `values` field over what is specified in the files.